prow/cmd/sinker/main.go

```diff
@@ -137,21 +137,28 @@ func main() {
 		logrus.WithError(err).Fatal("Error creating manager")
 	}
 
-	buildClusterClients, err := o.kubernetes.BuildClusterClients(cfg().PodNamespace, o.dryRun.Value)
+	buildManagers, err := o.kubernetes.BuildClusterManagers(o.dryRun.Value,
+		func(o *manager.Options) {
+			o.Namespace = cfg().PodNamespace
+		},
+	)
 	if err != nil {
-		logrus.WithError(err).Fatal("Error creating build cluster clients.")
+		logrus.WithError(err).Error("Failed to construct build cluster managers. Is there a bad entry in the kubeconfig secret?")
 	}
 
-	podClients := map[string]podInterface{}
-	for cluster, client := range buildClusterClients {
-		podClients[cluster] = client
+	buildClusterClients := map[string]ctrlruntimeclient.Client{}
+	for clusterName, buildManager := range buildManagers {
+		if err := mgr.Add(buildManager); err != nil {
+			logrus.WithError(err).Fatal("Failed to add build cluster manager to main manager")
+		}
+		buildClusterClients[clusterName] = mgr.GetClient()
```

Comment:
I'm pretty sure this should be `buildManager.GetClient()` rather than `mgr.GetClient()`. It is still unclear to me why Openshift's Prow instance doesn't have issues with this, but k8s-prow does.

Reply:
Openshifts Prow instance did have the same issue, but for some reason our alert didn't get triggered

Reply:
Our `sinker` *was* deleting .. something? From Friday:

```
$ oc --context app.ci --namespace ci logs --tail=-1 --selector component=sinker | grep 'Deleted old completed pod' | jq .cluster --raw-output | sort | uniq
api.ci
app.ci
build01
build02
vsphere
```

Reply:
Yeah, as cole pointed out it did get a client for all clusters, however the client was always the client to the cluster the prowjobs are in but stored under all build clusters names

Reply:
Oh, and of course NotFound is not an error when we're deleting ... hahaha perfect.

Reply:
Sounds like we should maybe have a separate metric for NotFound so that we can alert if we can't find a significant fraction of the pods.